native/core/src/models/status/membership.rs

```diff
@@ -1,147 +1,131 @@
-use matrix_sdk::ruma::{
-    events::room::member::MembershipChange as RumaMembershipChange, OwnedMxcUri, OwnedUserId,
-};
+use matrix_sdk::ruma::{OwnedMxcUri, OwnedUserId};
 use serde::{Deserialize, Serialize};
 
-/// A simple representation of a change, containing old and new data.
+// ruma_events::room::member::change::Change doesn't support serialization
 #[derive(Clone, Debug, Serialize, Deserialize)]
-#[allow(clippy::exhaustive_structs)]
 pub struct Change<T> {
     /// The old data.
-    pub old: T,
+    pub old_val: T,
 
     /// The new data.
-    pub new: T,
+    pub new_val: T,
 }
 
-/// Translation of the membership change in `m.room.member` event.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum MembershipChangeType {
-    /// User joined the room.
-    Joined,
-
-    /// User left the room.
-    Left,
-
-    /// User was banned.
-    Banned,
-
-    /// User was unbanned.
-    Unbanned,
-
-    /// User was kicked.
-    Kicked,
-
-    /// User was invited.
-    Invited,
-
-    /// User was kicked and banned.
-    KickedAndBanned,
-
-    /// User accepted the invite.
-    InvitationAccepted,
-
-    /// User rejected the invite.
-    InvitationRejected,
+impl<T: PartialEq> Change<T> {
+    pub fn new(old_val: T, new_val: T) -> Option<Self> {
+        if old_val == new_val {
+            None
+        } else {
+            Some(Self { old_val, new_val })
+        }
+    }
+}
 
-    /// User had their invite revoked.
-    InvitationRevoked,
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ProfileChange {
+    user_id: OwnedUserId,
+    display_name_change: Option<Change<Option<String>>>,
+    avatar_url_change: Option<Change<Option<OwnedMxcUri>>>,
+}
 
-    /// User knocked.
-    Knocked,
+impl ProfileChange {
```

Comment:
very few of the newly added lines in this file are covered by tests. Please extend the tests to include these lines.